ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -287,10 +299,90 @@ func (r *RayJobReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 			}
 			r.Log.Info("shutdownAfterJobFinishes set to true, we will delete cluster",
 				"RayJob", rayJobInstance.Name, "clusterName", fmt.Sprintf("%s/%s", rayJobInstance.Namespace, rayJobInstance.Status.RayClusterName))
-			return r.deleteCluster(ctx, rayJobInstance)
+			_, err = r.deleteCluster(ctx, rayJobInstance)
+			if err != nil && !errors.IsNotFound(err) {
+				return ctrl.Result{RequeueAfter: RayJobDefaultRequeueDuration}, nil
+			}
+			_, err = r.deleteK8sJob(ctx, rayJobInstance)
+			if err != nil && !errors.IsNotFound(err) {
+				return ctrl.Result{RequeueAfter: RayJobDefaultRequeueDuration}, nil
+			}
 		}
 	}
-	return ctrl.Result{}, nil
+	return ctrl.Result{RequeueAfter: RayJobDefaultRequeueDuration}, nil
+}
+
+// getOrCreateK8sJob creates a Kubernetes Job for the Ray Job if it doesn't exist, otherwise return the existing one.
+func (r *RayJobReconciler) getOrCreateK8sJob(ctx context.Context, rayJobInstance *rayv1alpha1.RayJob) (string, error) {
+	jobName := rayJobInstance.Name
+	jobNamespace := rayJobInstance.Namespace
+
+	// Create a Job object with the specified name and namespace
+	job := &batchv1.Job{}
+	if err := r.Client.Get(ctx, client.ObjectKey{Namespace: jobNamespace, Name: jobName}, job); err != nil {
+		if errors.IsNotFound(err) {
+			submitterTemplate, err := r.getSubmitterTemplate(rayJobInstance)
+			if err != nil {
+				return "", err
+			}
+			return r.createNewK8sJob(ctx, rayJobInstance, submitterTemplate)
+		}
+
+		// Some other error occurred while trying to get the Job
+		return "", err
+	}
+
+	// Job already exists, instead of returning an error we return a "success"
+	return jobName, nil
+}
+
+// getSubmitterTemplate builds the submitter pod template for the Ray job.
+func (r *RayJobReconciler) getSubmitterTemplate(rayJobInstance *rayv1alpha1.RayJob) (v1.PodTemplateSpec, error) {
+	var submitter_template v1.PodTemplateSpec
```

Comment:
This does not appear to follow the coding style. `submitterTemplate`?

Reply:
Thanks, fixed 6287936